spinalcordtoolbox/flattening.py

```diff
@@ -58,7 +58,11 @@ def flatten_sagittal(im_anat, im_centerline, verbose):
         img_reg = transform.warp(img, tform)
         im_anat_flattened.data[:, :, iz] = img_reg
 
-    # change back to native orientation
+    # Change [-1, 1] values back to the original range ([min, max])
+    im_anat_flattened.data = (im_anat_flattened.data + 1) * (max_data - min_data) / 2
```

Comment:
Good news and bad news:
- This formula _is_ the inverse transformation of the original transformation.
- But the original transformation _doesn't_ map `[min, max]` to `[-1, 1]`.

I think the original should be:
```py3
im_anat_flattened.data = 2/(max_data - min_data) * (im_anat_flattened.data - (max_data + min_data)/2)
```

And then the reverse would be:
```py3
im_anat_flattened.data = (max_data - min_data)/2 * im_anat_flattened.data + (max_data + min_data)/2
```

Reply:
> But the original transformation doesn't map [min, max] to [-1, 1].

Ahhh, good catch! I think the current formula is only valid when `min == 0`, in which case it becomes:

```python
# original LOC
im_anat_flattened.data = 2 * im_anat_flattened.data / (max_data - min_data) - 1

# when min == 0
im_anat_flattened.data = 2 * im_anat_flattened.data / max_data - 1

# substeps:
im_anat_flattened.data /= max_data  # [0, max] -> [ 0, 1]
im_anat_flattened.data *= 2         # [0, 1]   -> [ 0, 2]
im_anat_flattened.data -= 1         # [0, 2]   -> [-1, 1]
```

That might explain how this formula was originally determined? However, if `min != 0`, then a shift is needed prior to the division step:

```python
# substeps:
im_anat_flattened.data -= min_data               # [min, max]   -> [ 0, max-min]
im_anat_flattened.data /= (max_data - min_data)  # [0, max-min] -> [ 0, 1]
im_anat_flattened.data *= 2                      # [0, 1]       -> [ 0, 2]
im_anat_flattened.data -= 1                      # [0, 2]       -> [-1, 1]
```

So, I think the original could be simplified to:

```python
# [min, max] -> [0, max-min] -> [0, 1] -> [0, 2] -> [-1, 1]
im_anat_flattened.data = ((im_anat_flattened.data - min_data) / (max_data - min_data) * 2) - 1
```